### ISSUE TITLE
EmptyLineSeparator does not report problem, if next method is documented using Javadoc

### ISSUE THREAD

**Issue by pbaranchikov** [checkstyle/checkstyle#2266]
Absence of empty line is not reported, if the bottom method has JavaDoc.
```java
    public static void nethodPrev() {
        ...
    }
    /**
     * Some Javadoc
     */
    public static void methodNext(String entityId) {
        ...
    }
```
If I remove the JavaDoc for methodNext(), violation is reported

**Comment by romani** [comment]
Thanks for reporting this, that also problem for fields. There are a lot of code that miss line separator in Checkstyle code.

**Comment by pbaranchikov** [comment]
@romani, I found, that in test inputs there are:
```java
	static {  //warn
        //empty static initializer
	}
	//separator blank line
	{
		//empty instance initializer
	}
```
So, is a single-line comment a legal separator?
[Google Java Style] (https://google.github.io/styleguide/javaguide.html#s4.6.1-vertical-whitespace) says
> A single blank line appears:

so, single-line comments are not legal separator for this check according to Google Style.
Should the check count single line comments for empty lines?

**Comment by romani** [comment]
That is mistake in Check, single line comment is not a an empty line.
